samples/authUserFlow.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function main(projectId, clientId, clientSecret, refreshToken) {
+  // [START bigquery_auth_user_flow]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'PROJECT ID'
+  // const clientId = 'CLIENT ID'
+  // const clientSecret = 'CLIENT SECRET'
+  // const refreshToken = 'REFRESH TOKEN'
+
+  const credentials = {
+    type: 'authorized_user',
```

Comment:
I'm not sure if we actually support these options? I've never seen them documented anywhere (so I definitely could be wrong) but looking at the TypeScript interfaces it looks like the only options we accept here are `client_email` and `private_key`.

Reply:
I thought we didn't either! That's why I had this whole work-around when I first made the sample, but I tried it out based off a comment by @stephenplusplus and it appears to be working. I could be totally missing something, however, so please try it out and let me know, if you have time.

Reply:
I followed this all the way down the rabbit hole, and I actually think it would work.  The code in `googleauth.ts` is not good.  
https://github.com/googleapis/google-auth-library-nodejs/blob/master/src/auth/googleauth.ts#L448

If you follow that path, it should actually pull in the JSON you're passing here.